autopts/wid/mics.py

```diff
@@ -0,0 +1,103 @@
+#
+# auto-pts - The Bluetooth PTS Automation Framework
+#
+# Copyright (c) 2023, Codecoup.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms and conditions of the GNU General Public License,
+# version 2, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+import logging
+import sys
+
+from autopts.ptsprojects.stack import get_stack
+from autopts.pybtp import btp
+from autopts.pybtp import defs
+from autopts.pybtp.types import WIDParams
+
+log = logging.debug
+mute_state = None
+
+
+def mics_wid_hdl(wid, description, test_case_name, logs=True):
+    if logs:
+        log(f'{mics_wid_hdl.__name__}, {wid}, {description}, {test_case_name}')
+    module = sys.modules[__name__]
+
+    try:
+        handler = getattr(module, f'hdl_wid_{wid}')
+        return handler(WIDParams(wid, description, test_case_name))
+    except AttributeError as e:
+        logging.exception(e)
+
+
+# wid handlers section begin
+def hdl_wid_20001(_: WIDParams):
+    """Please prepare IUT into a connectable mode. Verify that the
+    Implementation Under Test (IUT) can accept GATT connect request from PTS."""
+    stack = get_stack()
+    btp.gap_set_conn()
+    btp.gap_adv_ind_on(ad=stack.gap.ad)
+    return True
+
+
+def hdl_wid_1(_: WIDParams):
+    """Please make sure IUT has set its Mute state to Disabled (0x02)"""
+    stack = get_stack()
+    btp.mics_mute_read()
+    mute = stack.mics.event_queues[defs.MICS_MUTE_STATE_EV][0][0]
+    ev = stack.mics.wait_mute_state_ev(30, remove=True)
+    if mute == 2 and ev:
+        return True
+
+    btp.mics_mute_disable()
+    ev = stack.mics.wait_mute_state_ev(30, remove=False)
+    mute = stack.mics.event_queues[defs.MICS_MUTE_STATE_EV][0][0]
+    if not ev and mute != 2:
+        return False
+
+    return True
+
+
+def hdl_wid_5(_: WIDParams):
+    """
+    Please update the Mute characteristic to 0 or 1
+    (whichever is different than the value in step 1) then send notification
+    of the Mute characteristic.
+    """
+    global mute_state
+    stack = get_stack()
+    if mute_state == 0:
+        btp.mics_mute()
+    else:
+        btp.mics_unmute()
+
+    mute_state = None
+    ev = stack.mics.wait_mute_state_ev(30, remove=False)
+    if ev is None:
+        return False
+
+    return True
+
+
+def hdl_wid_6(_: WIDParams):
+    """Please make sure the Mute characteristic is initially set to 0 or 1."""
+    global mute_state
+    stack = get_stack()
+    btp.mics_mute_read()
+    ev = stack.mics.wait_mute_state_ev(30, remove=False)
+    if ev is None:
+        return False
+
+    mute_state = stack.mics.event_queues[defs.MICS_MUTE_STATE_EV][0][0]
+
+    if mute_state not in range(0, 2):
```

Comment:
more transparent, imo
```suggestion
    if mute_state not in [0, 1]:
```